src/Shared/Helpers/OutputBuilder/StringOutputBuilder.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.CST.OpenSource.Shared
+{
+    class StringOutputBuilder : IOutputBuilder
+    {
+        StringBuilder stringResults = new StringBuilder();
+
+        /// <summary>
+        /// Prints to the currently selected output
+        /// </summary>
+        public void PrintOutput()
+        {
+            Console.Out.Write(this.GetOutput());
+        }
+
+        public string? GetOutput()
+        {
+            return stringResults.ToString();
+        }
+        /// <summary>
+        /// Append more text to the result
+        /// </summary>
+        /// <param name="output"></param>
+        public void AppendOutput(object? output)
+        {
+            var results = (IEnumerable<string>?)output ?? Array.Empty<String>().ToList();
```

Comment:
Why don't you specify that this takes IEnumerable<string> in the method declaration? This will throw an exception if you pass it other kinds of objects.

Reply:
makes sense...I started with adding both object and IEnumerable objects overloads, and ended up deleting the wrong one